src/main/java/seedu/mtracker/console/EditInstrumentParser.java

```diff
@@ -118,7 +119,20 @@ public static void editExitPriceParameter(HashSet<String> parametersGiven)
         editedParameters.put(EXIT_PRICE_ATTRIBUTE, inputExitPrice);
     }
 
-    public static void editExpiryParameter(HashSet<String> parametersGiven)
+    public void editDoneStatus(HashSet<String> parametersGiven) throws OperationAbortedError {
+        if (!parametersGiven.contains(DONE_ATTRIBUTE)) {
+            return;
+        }
+        String inputStatus;
+        do {
+            TextUi.displayEditStatus();
+            inputStatus = getUserInput(WORKSPACE).toLowerCase();
+            checkIfAbort(inputStatus, WORKSPACE);
```

Comment:
This is good, I apologise for not catching the rest earlier, but for the rest of editparameters method can shift the TextUi.displayEditXYZ() into the do while loop? Right now its just querying the edit parameter forever but the user don't see the instruction

Reply:
Same for the do while loop in add instrument type. Currently TextUi.displayAddInstrumentFirstInstruction(); it is only displayed once. Can help shift it inside the do while loop